### PR TITLE
Feature/pk service request header

### PR DESCRIPTION
**Pull request by hannakortetjarvi**

### CODE REVIEW

**Review by bunjovis — CHANGES_REQUESTED** [review]
This is not working for me. I had to add 
`import dotenv from 'dotenv';

dotenv.config();`
to cc-pk-service/app.ts and then it started working. Odd that you're not even touching that part in app.ts in this pull request, maybe dotenv worked fine because it was included in another file?

**Comment by hannakortetjarvi** [comment]
Noticed eslint files have disappeared(?) so 'npm run check' isn't working atm. I'm working on that

**Review by bunjovis — CHANGES_REQUESTED** [review]
In your own routes, also use the standard headers for token. meaning, the correct header where the token is should be Authorization, and the value should be "Bearer <token>"

File: cc-pk-service/src/routes/pkServiceRoutes.ts [hunk 1]
```diff
@@ -14,33 +15,39 @@ pkRouter.get('/system/ping', (_:Request, res:Response) => {
...
 // Route which returns user's items
 // http://localhost:5123/v1/users/:usedId/items
-pkRouter.get('/users/:userId/items', async (req:Request, res:Response) => {
-  if (req.params.userId === userId) {
-    const userSurveys:Survey[] = await getSurveys();
-    const reusableItems:Item[] = await getReusables(userSurveys);
+pkRouter.get('/users/:userId/items', async (req:Request, res:Response, next:NextFunction) => {
+  req.header('Token');
+  try {
+    const token:any = req.headers.token ?? '';
```
Comment (OpaJaa) [inline cc-pk-service/src/routes/pkServiceRoutes.ts]: Maybe change to req.headers.authorization, then split the token as "Bearer" is left in it. Sent you code snippets to check.

File: cc-pk-service/src/handleData.ts [hunk 2]
```diff
@@ -1,121 +1,74 @@
...
-                'grant_type': g_type,
-                'username': username,
-                'password': password
-        }));
-        return (await response).data.access_token;
-    } catch (err) {
-        console.log('Error: cannot fetch id for the current user');
-        return err;
-    }
-}
```
Comment (OpaJaa) [inline cc-pk-service/src/handleData.ts]: If doing a new getToken -function for splitting and returning, you could add it here. Code snippet also from this in teams chat

**Review by OpaJaa — CHANGES_REQUESTED** [review]
Sent my version of resolving tokens in teams chat to you. If suggestions, can change my implementation also. But either way me or you or both have to do changes :)

File: cc-pk-service/.env.example [hunk 3]
```diff
@@ -1,5 +1,17 @@
...
+PK_CLIENTID="{CLIENT ID FROM THE CURL REQUEST}"
+PK_GRANTTYPE="{GRANT TYPE FROM THE CURL REQUEST}"
+=======
 PORT=5123
 PK_USERNAME="{YOUR USERNAME FOR PURKUKARTOITUS}"
 PK_PASSWORD="{YOUR PASSWORD FOR PURKUKARTOITUS}"
 PK_CLIENTID="{CLIENT ID FROM THE CURL REQUEST}"
 PK_GRANTTYPE="{GRANT TYPE FROM THE CURL REQUEST}"
+>>>>>>> 690411f799daee607b8d1d21a860e6987ff7b605
+>>>>>>> main
```
Comment (OpaJaa) [inline cc-pk-service/.env.example]: I think this has the same conflicts remaining as was in my repo. Should remove those 👍

File: cc-pk-service/.env.example [hunk 4]
```diff
@@ -1,5 +1,17 @@
+<<<<<<< HEAD
```
Comment (bunjovis) [inline cc-pk-service/.env.example]: Fix merge conflicts in this file

**Review by bunjovis — CHANGES_REQUESTED** [review]

**Comment by MSurvonen** [comment]
I had to write npm install cors and started to work after that

**Review by bunjovis — APPROVED** [review]

**Review by OpaJaa — APPROVED** [review]

**Pull request merged**